src/Microsoft.Data.Sqlite.Core/SqliteConnection.CreateAggregate.cs

```diff
@@ -943,6 +943,995 @@ public virtual void CreateAggregate<TAccumulate>(string name, TAccumulate seed,
         ///     delete a function.
         /// </param>
         public virtual void CreateAggregate<TAccumulate, TResult>(string name, TAccumulate seed, Func<TAccumulate, object[], TAccumulate> func, Func<TAccumulate, TResult> resultSelector)
-            => CreateAggregateCore(name, -1, seed, IfNotNull<TAccumulate, TAccumulate>(func, (a, r) => func(a, GetValues(r))), resultSelector);
+            => CreateAggregateCore(name, -1, seed, IfNotNull<TAccumulate, TAccumulate>(func, (a, r) => func(a, GetValues(r))), resultSelector, false);
+
+        /// <summary>
+        ///     Creates or redefines an aggregate SQL function.
+        /// </summary>
+        /// <typeparam name="TAccumulate">The type of the accumulator value.</typeparam>
+        /// <param name="name">The name of the SQL function.</param>
+        /// <param name="func">An accumulator function to be invoked on each element. Pass null to delete a function.</param>
+        /// <param name="isDeterministic">Flag indicating whether the aggregate is deterministic.</param>
+        public virtual void CreateAggregate<TAccumulate>(string name, Func<TAccumulate, TAccumulate> func, bool isDeterministic)
```

Comment:
We should consider just giving the `isDeterministic` parameter a default value of `false` instead of having two overloads.